### PR TITLE
Milestone 8: Dynamic modal and popup

### PR DESCRIPTION
**Pull request by shahedakhatoon**
## Added the following:
1. Works section dynamically using javascript.
2. Popup window.
3. Linters for html+css+javascript.

### CODE REVIEW

File: scripts/dynamicmodal.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      sourceLinkDom.action = projectsArray[i].sourceLink;
+      demoLinkDom.action = projectsArray[i].demoLink;
+      technologiesDom.innerHTML = '';
+      for (let j = 0; j < projectsArray[i].tags.length; j += 1) {
+        technologiesDom.innerHTML = `${technologiesDom.innerHTML}<li class="tag">${projectsArray[i].tags[j]}</li>`;
+      }
+      popupModal.classList.remove('vanish');
+      popupModal.classList.add('show');
+      mobileMenu.classList.add('blur');
+      logo.classList.add('blur');
```
Comment (AnselemOdims) [inline scripts/dynamicmodal.js:177]: - [ ] Please kindly ensure the popup modal looks very similar to the original design on [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1). At the moment, the text and buttons are overflowing the modal. Kindly compare your design with that of the original design below.

### Your Design
image

### Original Design
image

File: scripts/dynamicmodal.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    });
+    viewBtn[i].index = i;
+  }
+})();
+
+closeModal.addEventListener('click', () => {
+  popupModal.classList.remove('show');
+  popupModal.classList.add('vanish');
+  mobileMenu.classList.remove('blur');
```
Comment (AnselemOdims) [inline scripts/dynamicmodal.js:186]: - [ ] Kindly also ensure the mobile view matches the design on [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1)

### Your Design
image

### Original Design
image

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @shahedakhatoon , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: scripts/dynamicmodal.js (hunk 2 above)
Comment (micheaol) [reply scripts/dynamicmodal.js:186]: As requested by the previous code reviewer, kindly ensure mobile view matches the design on [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1)

Your Implementation 
image

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @shahedakhatoon , 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: scripts/dynamicmodal.js (hunk 2 above)
Comment (shahedakhatoon) [reply scripts/dynamicmodal.js:186]: @micheaol Please see the screenshots I've provided, I think the desktop and mobile versions are now perfect. 
Desktop version
Desktop
Mobile version
Mobile

Comment (micheaol) [reply scripts/dynamicmodal.js:186]: @shahedakhatoon ,

Kindly re-submit for review on your dashboard

Comment (shahedakhatoon) [reply scripts/dynamicmodal.js:186]: @micheaol okay, thank you.

**Review by Kingstalux — APPROVED** [review]
Hi @shahedakhatoon ,
# Status: Approved ✔️ ✔️ ✔️ 
 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**